Client/mods/deathmatch/logic/lua/CLuaArguments.cpp

```diff
@@ -535,46 +535,51 @@ json_object* CLuaArguments::WriteTableToJSONObject(bool bSerialize, CFastHashMap
 
     pKnownTables->insert(std::make_pair(this, pKnownTables->size()));
 
-    bool                                  bIsArray = true;
-    unsigned int                          iArrayPos = 1;            // lua arrays are 1 based
-    vector<CLuaArgument*>::const_iterator iter = m_Arguments.begin();
+    bool                                      bIsArray = true;
+    vector<pair<unsigned int, CLuaArgument*>> vecSortedArguments;            // lua arrays are not necessarily sorted
```

Comment:
`std::uint32_t` instead of `unsigned int`

Also use `std::vector` instead of `vector`.
`using namespace std` should be banned